feature/experimental/gribi/ate_tests/supervisor_failure/Supervisor_Failure_test.go

```diff
@@ -0,0 +1,347 @@
+	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandbyBeforeSwitch, rpActiveBeforeSwitch)
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: &tpb.Path{
+			Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
+		},
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+	switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+	if err != nil {
+		t.Fatalf("Failed to perform control processor switchover with unexpected err: %v", err)
+	}
+	t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+
+	rpStandbyAfterSwitch, rpActiveAfterSwitch := findStandbyRP(t, dut, supervisors)
+	t.Logf("Found standbyRP after switchover: %v, activeRP: %v", rpStandbyAfterSwitch, rpActiveAfterSwitch)
+	t.Log("Validate OC Switchover time/reason.")
+	activeRP := dut.Telemetry().Component(rpActiveAfterSwitch)
+	if !activeRP.LastSwitchoverTime().Lookup(t).IsPresent() {
+		t.Errorf("activeRP.LastSwitchoverTime().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		t.Logf("Found activeRP.LastSwitchoverTime(): %v", activeRP.LastSwitchoverTime().Get(t))
+	}
+
+	if !activeRP.LastSwitchoverReason().Lookup(t).IsPresent() {
+		t.Errorf("activeRP.LastSwitchoverReason().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastSwitchoverReason := activeRP.LastSwitchoverReason().Get(t)
+		t.Logf("Found lastSwitchoverReason.GetDetails(): %v", lastSwitchoverReason.GetDetails())
+		t.Logf("Found lastSwitchoverReason.GetTrigger().String(): %v", lastSwitchoverReason.GetTrigger().String())
+	}
+
+	if !activeRP.LastRebootTime().Lookup(t).IsPresent() {
+		t.Errorf("activeRP.LastRebootTime.().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastrebootTime := activeRP.LastRebootTime().Get(t)
+		t.Logf("Found lastRebootTime.GetDetails(): %v", lastrebootTime)
+	}
+	if !activeRP.LastRebootReason().Lookup(t).IsPresent() {
+		t.Errorf("activeRP.LastRebootReason.().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastrebootReason := activeRP.LastRebootReason().Get(t)
+		t.Logf("Found lastRebootReason.GetDetails(): %v", lastrebootReason)
+	}
+
+	// Assume Supervisor Switchover happened, ensure traffic flows again.
+	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
+	defer clientA.Close(t)
+	if err := clientA.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection can not be established")
+	}
+	ipv4Path := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+	}
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+	testTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
```

Comment:
Can we start traffic before the switchover and then continue until some time after switchover. This would give us a fair idea whether there were any disruptions of traffic during the switchover.

Reply:
ok. Will keep traffic running like below. 

 Start traffic  , ensure no packets loss
 do supervisor switchover
 Check traffic and ensure no packet loss.

Reply:
Implemented.. New code push failed due to "Needs /gcbrun from a collaborator"..